.github/workflows/main.yml

```diff
@@ -32,7 +32,15 @@ jobs:
         run: |
           VERSIONS=$(parse-dockerfile Dockerfile | jq -cr '[.instructions[] | select(.kind=="LABEL" or .kind=="ARG") | select(.arguments.value | contains("org.opencontainers.image") or startswith("APOLLO_")).arguments.value | match("([^=]*)=(.*)") | .captures | {(.[0].string) : .[1].string}] | add')
           echo "versions=$VERSIONS" >> "$GITHUB_OUTPUT"
-      - name: Check Image to Build Post-Merge Does Not Already Exist
+          
+          # Check if preview version is defined as workflow input
+          if [ -n "${{ inputs.preview_router_version }}" ]; then
+            echo "has_preview=true" >> "$GITHUB_OUTPUT"
+            echo "preview_version=${{ inputs.preview_router_version }}" >> "$GITHUB_OUTPUT"
+          else
+            echo "has_preview=false" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
I think, if we can use the `docker/metadata-action` we probably don't need to do this. You can use the tags field there to set all sorts of conditonal behaviour: https://github.com/docker/metadata-action?tab=readme-ov-file#tags-input